darwin/future/meta/queries/workflow.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime, timezone
+from typing import List
+from uuid import UUID
+
+from darwin.exceptions import DarwinException
+from darwin.future.core.client import Client
+from darwin.future.core.types.query import Param, Query, QueryFilter
+from darwin.future.core.workflows.list_workflows import list_workflows
+from darwin.future.data_objects.workflow import WFStage, Workflow
+from darwin.future.helpers.exception_handler import handle_exception
+from darwin.future.meta.objects.workflow import WorkflowMeta
+
+
+class WorkflowQuery(Query[WorkflowMeta, Workflow]):
+    """
+    WorkflowQuery object with methods to manage filters, retrieve data, and execute
+    filters
+
+    Methods
+    -------
+
+    where: Adds a filter to the query
+    collect: Executes the query and returns the filtered data
+    """
+
+    def where(self, param: Param) -> "WorkflowQuery":
+        filter = QueryFilter.parse_obj(param)
+        query = self + filter
+
+        return WorkflowQuery(self.client, query.filters)
+
+    def collect(self) -> List[Workflow]:
+        workflows, exceptions = list_workflows(self.client)
+        if exceptions:
+            handle_exception(exceptions)
+            raise DarwinException from exceptions[0]
+
+        if not self.filters:
+            return workflows
+
+        for filter in self.filters:
+            workflows = self._execute_filters(workflows, filter)
+
+        return workflows
+
+    def _execute_filters(self, workflows: List[Workflow], filter: QueryFilter) -> List[Workflow]:
```

Comment:
I know this is my devising but for these complex objects I'm less enamoured with this filtering approach... but I can't think of anything better that still allows for the customizability for all the different (and custom) types we will have. Open to any suggestions if you've got them.

Reply:
Best I can think of is maybe a member taking a list of filtering functions?

Reply:
The order should be reversed, so we only iterate over the list once rather than once per filter. Doesn't really matter here but for big datasets for instance it might become an issue. With the inheritance and typing though it can easily be implemented later for objects that need it. Just have to overwrite (or not even use) generic_execute_filter with one that is O(n) instead. 

Maybe filters becomes a list of functions that get called so something like 

`filtered = [x for x in unfiltered if any([y(x) for y in filters])]`

is possible